### PR TITLE
Fixes #34913: azure_rm_loadbalance should accpet a list of rules instead of one

### LINKED ISSUE #34913: azure_rm_loadbalance should accpet a list of rules instead of one
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME
azure_rm_loadbalance

##### ANSIBLE VERSION
2.4
2.5

##### CONFIGURATION
N/A

##### OS / ENVIRONMENT
N/A

##### SUMMARY
The load balancer module. Currently it can only accept one load balancing rule and one NAT rule.  It should be able to accept a list of rules.

### PR DESCRIPTION
**Pull request by yuwzho**
##### SUMMARY

Fixes #34913

##### ISSUE TYPE

 - Feature Pull Request

##### ANSIBLE VERSION

```
2.4
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py [hunk 1]
```diff
@@ -61,89 +61,248 @@
...
+                required: True
+            backend_address_pool:
+                description: A reference to a pool of DIPs. Inbound traffic is randomly load balanced across IPs in the backend IPs.
+                required: True
+            probe:
+                description: The reference of the load balancer probe used by the load balancing rule.
+                required: True
+            protocol:
+                description: "Possible values include: 'Udp', 'Tcp', 'All'."
+                choices:
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py]: should replace 'Udp' with C(udp), etc...

File: lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py [hunk 2]
```diff
@@ -61,89 +61,248 @@
...
+                    - Tcp
+                    - Udp
+                    - All
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py]: also the recent rule is to have choice values in snake_lower_case format

File: lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py [hunk 3]
```diff
@@ -296,177 +597,204 @@ def __init__(self):
...
+        load_balancer = self.get_load_balancer()
 
-        if self.probe_protocol:
-            load_balancer_props['probes'] = [
-                self.network_models.Probe(
-                    name=probe_name,
-                    protocol=self.probe_protocol,
+        if self.state == 'present':
+            # compatiable parameters
+            if not self.frontend_ip_configurations and not self.backend_address_pools and not self.probes and not self.inbound_nat_pools:
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py]: compatiable?

File: lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py (hunk 2 above)
Comment (yuwzho) [reply lib/ansible/modules/cloud/azure/azure_rm_loadbalancer.py]: These are strings accepted by SDK.

**Comment by zikalino** [comment]
shipit

**Comment by zikalino** [comment]
bot_status

**Comment by xscript** [comment]
shipit

**Comment by zikalino** [comment]
bot_status

**Comment by nitzmahone** [comment]
Apparently it didn't merge because it touches files outside the module. It's already an open enhancement request- we definitely want people to be able to automerge things that just touch their own tests. https://github.com/ansible/ansibullbot/issues/808

**Pull request merged**